packages/customWidgets/calendar-web/src/components/CalendarContainer.ts

```diff
@@ -308,11 +325,24 @@ export default class CalendarContainer extends Component<Container.CalendarConta
             datePattern = dateFormat || "EEEE";
         } else if (dateType === "timeGutter") {
             datePattern = dateFormat || "hh:mm a";
+        } else {
+            datePattern = dateFormat;
         }
 
         return (date: Date) => window.mx.parser.formatValue(date, "datetime", { datePattern });
     };
 
+    private customRangeFormat = (dateFormat: string): ((dateRange: { start: Date; end: Date }) => string) => {
+        const datePattern = dateFormat;
+        return (dateRange: { start: Date; end: Date }) => {
+            console.log("start", dateRange.start);
+            console.log("end", dateRange.end);
+            return `${window.mx.parser.formatValue(dateRange.start, "datetime", {
```

Comment:
is it safe?

Reply:
Yes, `start` and `end` should always be of type Date. The API docs are inconclusive about this however. I had to search a bit for the required callback function definition. The same counts btw for the `customFormat` function above, which was already implemented.